src/test_fixture/sharing.rs

```diff
@@ -170,12 +170,34 @@ impl<F: Field> Reconstruct<F> for [&Replicated<F>; 3] {
     }
 }
 
+impl<F: Field> Reconstruct<F> for [&MaliciousReplicated<F>; 3] {
+    fn reconstruct(&self) -> F {
+        use crate::secret_sharing::ThisCodeIsAuthorizedToDowngradeFromMalicious;
+        let s0 = self[0];
+        let s1 = self[1];
+        let s2 = self[2];
+        [s0.rx(), s1.rx(), s2.rx()].reconstruct();
```

Comment:
Note that this isn't really needed.  It only validates that the shares are sensible, not that this is `r` times `x`.

Reply:
Removing this since not needed anymore